src/main/java/com/playmonumenta/networkchat/RemotePlayerListener.java

```diff
@@ -181,7 +181,13 @@ public void gatherRemotePlayerDataEvent(GatherRemotePlayerDataEvent event) {
 
 		JsonObject playerJson = new JsonObject();
 
-		Component component = MessagingUtils.playerComponent(player);
+		Component component;
+		if (event.mRemotePlayer.isHidden() != null && event.mRemotePlayer.isHidden()) {
```

Comment:
This might need to be `isHidden != false`; if I remember right, null/undefined defaults to hidden, otherwise treat it as correct. Solves an edge case with players transferring shards while also "online" through something like a Discord bridge.

Reply:
`event.mRemotePlayer.isHidden() == false` would work right?